vyper/codegen/core.py

```diff
@@ -290,6 +290,65 @@ def append_dyn_array(darray_node, elem_node):
             return IRnode.from_list(b1.resolve(b2.resolve(ret)))
 
 
+def extend_dyn_array(context, dst_darray_node, src_darray_node):
+    assert isinstance(dst_darray_node.typ, DArrayType)
+    assert isinstance(src_darray_node.typ, DArrayType)
+
+    ret = ["seq"]
+    with dst_darray_node.cache_when_complex("dst_darray") as (
+        b1,
+        dst_darray_node,
+    ), src_darray_node.cache_when_complex("src_darray") as (b2, src_darray_node):
+        dst_len = get_dyn_array_count(dst_darray_node)
+        src_len = get_dyn_array_count(src_darray_node)
+        max_dst_len = dst_darray_node.typ.count
+
+        with dst_len.cache_when_complex("dst_darray_len") as (
+            b3,
+            dst_len,
+        ), src_len.cache_when_complex("src_darray_len") as (b4, src_len):
+            # Assert that dst darray is not at maximum length
+            dst_bounds_assertion = ["assert", ["lt", dst_len, max_dst_len]]
+            ret.append(dst_bounds_assertion)
+
+            # Assert len(src_darray) + len(dst_darray) < maxlen(dst_darray)
+            combined_len = ["add", dst_len, src_len]
+            within_maxlen_assertion = ["assert", ["le", combined_len, max_dst_len]]
+            ret.append(within_maxlen_assertion)
+
+            loop_body = ["seq"]
+
+            # Loop over source darray until it runs out
+            loop_var = IRnode.from_list(context.fresh_varname("dynarray_extend_ix"), typ="uint256")
+            iter_count = IRnode.from_list(["sub", max_dst_len, dst_len])
+
+            # Copy element from src darray to dst darray and increment length
+            # Index of src darray to copy is `current iteration - original length of dst darray`
+            src_idx = IRnode.from_list(["sub", loop_var, dst_len], typ="uint256")
```

Comment:
i think src_idx can actually just be `loop_var`?

Reply:
I think src_idx is needed because we are copying from index 0 of src darray to the index representing the current length of the dst darray.